schematic/lepton-schematic.scm

```diff
@@ -265,42 +179,146 @@ Run `~A --help' for more information.\n")
     '())))
 
 
-(define main
-  (pointer->procedure
-   '*
-   (dynamic-func "main_prog" libleptongui)
-   (list '*)))
+;;; Load GTK resource files.
+;;; Search system and user configuration directories for
+;;; lepton-gtkrc files and load them in sequence.
+(define (parse-gtkrc)
+  (let loop ((dirs (append (sys-config-dirs)
+                           (list (user-config-dir)))))
+    (or (null? dirs)
+        (let ((filename (string-append (car dirs)
+                                       file-name-separator-string
+                                       "lepton-gtkrc")))
+          (when (file-readable? filename)
+            (gtk_rc_parse filename))
+          (loop (cdr dirs))))))
+
+
+
+
+;;; Setup default icon for GTK windows
+;;; Sets the default window icon by name, to be found in the
+;;; current icon theme.
+(define (set-window-default-icon)
+  (define %theme-icon-name "lepton-schematic")
+
+  (gtk_window_set_default_icon_name %theme-icon-name))
+
+;;; Setup icon search paths.
+;;; Add the icons installed by the program to the search path for
+;;; the default icon theme, so that they can be automatically
+;;; found by GTK.
+(define (init-window-icons)
+  ;; FIXME: this shouldn't be necessary, Lepton should just
+  ;; install its icons in the system hicolor icon theme and
+  ;; they'll be picked up automatically.
+  (let loop ((sys-dirs (sys-data-dirs)))
+    (or (null? sys-dirs)
+        (let ((icon-dir (string-append (car sys-dirs)
+                                       file-name-separator-string
+                                       "icons")))
+          (gtk_icon_theme_append_search_path (gtk_icon_theme_get_default)
+                                             icon-dir)
+          (loop (cdr sys-dirs))))))
+
+
+(define (open-log-window)
+  (let ((cfg (path-config-context (getcwd))))
+    (when (string= (config-string cfg "schematic" "log-window")
+                   "startup")
+      (x_widgets_show_log window))))
```

Comment:
`window` is undefined here.
If `schematic::log-window` is set to "startup", lepton-schematic exits with:
```
Backtrace:
           5 (primitive-load "/home/dmn/lepton/bin.tb/bin/lepton-sch…")
In ice-9/eval.scm:
   293:34  4 (_ #(#<directory (guile-user) 800731140> ()))
    619:8  3 (_ #(#(#(#<directory (guile-user) 800731140>) ()) #<…> …))
    155:9  2 (_ #(#(#(#<directory (guile-user) 800731140>)) #<geda…>))
   223:20  1 (proc #(#(#(#<directory (guile-user) 800731140>)) #<g…>))
In unknown file:
           0 (%resolve-variable (7 . window) #<directory (guile-user…>)

ERROR: In procedure %resolve-variable:
Unbound variable: window
```
The attached patch fixes this.

[log_window_on_startup.diff.txt](https://github.com/lepton-eda/lepton-eda/files/5741942/log_window_on_startup.diff.txt)

Reply:
Thanks, I overlooked this during refactoring.